lib/auth/auth_with_roles.go

```diff
@@ -1135,45 +1136,42 @@ func (a *ServerWithRoles) ListResources(ctx context.Context, req proto.ListResou
 	req.SearchKeywords = nil
 	req.PredicateExpression = ""
 
-	var resources []types.ResourceWithLabels
-
 	resourceChecker, err := a.newResourceAccessChecker(req.ResourceType)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 
-	resp, err := a.authServer.IterateResourcePages(ctx, req, func(nextPage []types.ResourceWithLabels) (bool, error) {
-		for _, resource := range nextPage {
-			if len(resources) == limit {
-				break
-			}
-
-			if err := resourceChecker.CanAccess(resource); err != nil {
-				if trace.IsAccessDenied(err) {
-					continue
-				}
+	resp := &types.ListResourcesResponse{
+		Resources: make([]types.ResourceWithLabels, 0, limit),
+	}
```

Comment:
I think we should cap the amount of preallocation we're willing to do.  Wouldn't want someone to be able to deliberately OOM the auth server by sending requests with giant limit values.

Reply:
I was going to say something about it only messing up some GC stats, but at least on macOS a `make([]byte, 0, 1<<40)` does seem to actually try to zero out a TiB of ram 😔

Should we just not preallocate? The usual reallocation strategy should get to an appropriate size pretty quickly even starting from 0.